src/useStorage/index.ts

```diff
@@ -0,0 +1,245 @@
+  /**
+   * Watch for deep changes
+   *
+   * @default true
+   */
+  deep?: boolean;
+
+  /**
+   * Listen to storage changes, useful for multiple tabs application
+   *
+   * @default true
+   */
+  listenToStorageChanges?: boolean;
+
+  /**
+   * Write the default value to the storage when it does not exist
+   *
+   * @default true
+   */
+  writeDefaults?: boolean;
+
+  /**
+   * Merge the default value with the value read from the storage.
+   *
+   * When setting it to true, it will perform a **shallow merge** for objects. You can pass a
+   * function to perform custom merge (e.g. deep merge), for example:
+   *
+   * @default false
+   */
+  mergeDefaults?: boolean | ((storageValue: T, defaults: T) => T);
+
+  /** Custom data serialization */
+  serializer?: Serializer<T>;
+
+  /**
+   * On error callback
+   *
+   * Default log error to `console.error`
+   */
+  onError?: (error: unknown) => void;
+
+  /**
+   * Use shallow ref as reference
+   *
+   * @default false
+   */
+  shallow?: boolean;
+
+  /**
+   * Wait for the component to be mounted before reading the storage.
+   *
+   * @default false
+   */
+  initOnMounted?: boolean;
+}
+
+export function useStorage(
+  key: string,
+  defaults: MaybeRefOrGetter<string>,
+  options?: UseStorageOptions<string>,
+): RemovableRef<string>;
+export function useStorage(
+  key: string,
+  defaults: MaybeRefOrGetter<boolean>,
+  options?: UseStorageOptions<boolean>,
+): RemovableRef<boolean>;
+export function useStorage(
+  key: string,
+  defaults: MaybeRefOrGetter<number>,
+  options?: UseStorageOptions<number>,
+): RemovableRef<number>;
+export function useStorage<T>(
+  key: string,
+  defaults: MaybeRefOrGetter<T>,
+  options?: UseStorageOptions<T>,
+): RemovableRef<T>;
+export function useStorage<T = unknown>(
+  key: string,
+  defaults: MaybeRefOrGetter<null>,
+  options?: UseStorageOptions<T>,
+): RemovableRef<T>;
+
+export function useStorage<T extends string | number | boolean | object | null>(
+  key: string,
+  defaults: MaybeRefOrGetter<T>,
+  options: UseStorageOptions<T> = {},
+): RemovableRef<T> {
+  const {
+    flush = 'pre',
+    deep = true,
+    listenToStorageChanges = true,
+    writeDefaults = true,
+    mergeDefaults = false,
+    shallow,
+    eventFilter,
+    onError = (e) => console.error(e),
+    initOnMounted,
+  } = options;
+  const data = (shallow ? shallowRef : ref)(typeof defaults === 'function' ? defaults() : defaults);
+
+  const storage = UniStorage;
+
+  const rawInit = resolveUnref(defaults);
+  const type = guessSerializerType(rawInit);
+  const serializer = options.serializer ?? StorageSerializers[type];
+
+  const { pause: pauseWatch, resume: resumeWatch } = pausableWatch(data, () => write(data.value), {
+    flush,
+    deep,
+    eventFilter,
+  });
+
+  if (listenToStorageChanges) {
+    tryOnLoad(() => {
+      // this should be fine since we are in a mounted hook
+      if (initOnMounted) update();
+    });
+  }
```

Comment:
I don't think it actually listens to storage changes. Can you please explain it?

Reply:
Thanks, I removed `listenToStorageChanges`